app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt

```diff
@@ -188,67 +185,93 @@ fun CameraFeedbackButton(onClick: () -> Unit = {}) {
  * @param numbOfHeaders The number of headers at the top of the list, allowing for an offset when
  *   scrolling to the selected letter.
  */
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun LetterDictionary(
     scrollState: LazyListState,
     coroutineScope: CoroutineScope,
     numbOfHeaders: Int
 ) {
-  var currentLetterIndex by remember { mutableIntStateOf(0) }
   val letters = ('a'..'z').toList()
+  val pagerState = rememberPagerState(initialPage = 0, pageCount = { letters.size })
 
-  Row(
-      modifier = Modifier.fillMaxWidth().wrapContentHeight().testTag("LetterDictionary"),
-      horizontalArrangement = Arrangement.SpaceBetween,
-      verticalAlignment = Alignment.CenterVertically) {
-        IconButton(
-            onClick = {
-              currentLetterIndex = (currentLetterIndex - 1 + letters.size) % letters.size
-            },
-            modifier = Modifier.testTag("LetterDictionaryBack")) {
-              Icon(
-                  Icons.AutoMirrored.Outlined.ArrowBack,
-                  tint = MaterialTheme.colorScheme.onBackground,
-                  contentDescription = "Back")
-            }
+  Box(modifier = Modifier.fillMaxWidth().wrapContentHeight().testTag("LetterDictionary")) {
+    // Back Arrow at the start
+    IconButton(
+        onClick = {
+          coroutineScope.launch {
+            pagerState.animateScrollToPage(
+                (pagerState.currentPage - 1 + letters.size) % letters.size)
+          }
+        },
+        modifier = Modifier.align(Alignment.CenterStart).testTag("LetterDictionaryBack")) {
+          Icon(
+              Icons.AutoMirrored.Outlined.ArrowBack,
+              tint = MaterialTheme.colorScheme.onBackground,
+              contentDescription = "Back")
+        }
 
-        val currentLetter = letters[currentLetterIndex]
-        Box(
-            modifier =
-                Modifier.border(2.dp, MaterialTheme.colorScheme.outline, RoundedCornerShape(8.dp))
-                    .background(MaterialTheme.colorScheme.primary, RoundedCornerShape(8.dp))
-                    .padding(8.dp)
-                    .clickable {
-                      coroutineScope.launch {
-                        scrollState.animateScrollToItem(currentLetterIndex + numbOfHeaders)
+    // Horizontal Pager for letters in the center
+    HorizontalPager(
+        beyondBoundsPageCount = 1,
+        state = pagerState,
+        modifier = Modifier.size(300.dp, 50.dp).align(Alignment.Center).testTag("LetterPager"),
+    ) { page ->
```

Comment:
This is a great use of HorizontalPager for enabling swipe gestures. It's intuitive and complements the existing button navigation.